File: client/src/api/types/response/JoinChatAllowance.ts
export enum JoinChatAllowance {
    NOT_ALLOWED = "NOT_ALLOWED",
    REQUIRES_APPROVAL = "REQUIRES_APPROVAL",
    INVITE_ONLY = "INVITE_ONLY",
    ALLOWED = "ALLOWED"
}

export const JOIN_CHAT_ALLOWANCES = [
    JoinChatAllowance.NOT_ALLOWED,
    JoinChatAllowance.REQUIRES_APPROVAL,
    JoinChatAllowance.INVITE_ONLY,
    JoinChatAllowance.ALLOWED
];

File: chat/src/main/kotlin/chatox/chat/model/JoinChatAllowance.kt
package chatox.chat.model

import chatox.platform.security.VerificationLevel

enum class JoinChatAllowance {
    NOT_ALLOWED,
    REQUIRES_APPROVAL,
    INVITE_ONLY,
    ALLOWED;

    companion object {

        @JvmStatic
        fun getAllowance(
                verificationLevel: VerificationLevel,
                allowanceMap: Map<VerificationLevel, JoinChatAllowance>?
        ): JoinChatAllowance = allowanceMap?.get(verificationLevel) ?: ALLOWED
    }
}
